standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -1295,26 +1288,39 @@ private static boolean timedOut(CompactionInfo ci, RetentionCounters rc, long pa
             && (rc.hasSucceededMajorCompaction || (rc.hasSucceededMinorCompaction && ci.type == CompactionType.MINOR));
   }
 
-  private void removeAbortRetryEntries(Connection dbConn, CompactionInfo info) throws MetaException, RetryException {
-    LOG.debug("Going to execute update <{}>", DELETE_ABORT_RETRY_ENTRIES_FROM_COMPACTION_QUEUE);
-    try (PreparedStatement pStmt = dbConn.prepareStatement(DELETE_ABORT_RETRY_ENTRIES_FROM_COMPACTION_QUEUE)) {
-      pStmt.setString(1, info.dbname);
-      pStmt.setString(2, info.tableName);
-      if (info.partName != null) {
-        pStmt.setString(3, info.partName);
-      } else {
-        // Since the type of 'CQ_PARTITION' column is varchar.
-        // Hence, setting null for VARCHAR type.
-        pStmt.setNull(3, Types.VARCHAR);
+  private void removeCompactionAndAbortRetryEntries(Connection dbConn, CompactionInfo info) throws MetaException, RetryException {
+    // Do not perform delete when the related records do not exist.
+    // This is valid in case of no abort retry.
+    if (info.id > 0) {
+      String query = DELETE_COMPACTION_AND_ABORT_RETRY_ENTRIES_FROM_COMPACTION_QUEUE;
+      // Add clause to delete abort retry records when the request is compaction.
+      if (!info.isAbortedTxnCleanup()) {
+        query += " OR ( \"CQ_DATABASE\" = ? " +
+            " AND \"CQ_TABLE\" = ? " +
+            " AND \"CQ_PARTITION\" " + (info.partName != null ? " = ? " : " IS NULL ") +
+            " AND \"CQ_TYPE\" = " +
+            quoteChar(TxnStore.ABORT_TXN_CLEANUP_TYPE) + " )";
+      }
```

Comment:
fix the indentation, that curly bracket is not aligned with proper if block

Reply:
Not required since this is a separate String constant.